docs/examples/plot_diffuse_fraction.py

```diff
@@ -0,0 +1,186 @@
+"""
+Diffuse Fraction Estimation
+===========================
+
+Comparison of diffuse fraction estimation methods used to derive direct and
+diffuse components from measured global horizontal irradiance.
+"""
+
+# %%
+# This example demonstrates how to use diffuse fraction estimation methods to
+# obtain direct and diffuse components from measured global horizontal
+# irradiance (GHI). PV systems are often tilted to optimize performance, so the
+# entire diffuse sky dome may not be visible to the PV surface. Determining the
+# total irradiance incident on the plane of the array requires transposing the
+# diffuse component, but irradiance sensors such as pyranometers typically only
+# measure GHI. Therefore pvlib provides several correlations to estimate the
+# diffuse fraction of the GHI, that can be used to resolve the diffuse and
+# direct components.
+
+import pathlib
+from matplotlib import pyplot as plt
+import numpy as np
+import pandas as pd
+from pvlib.iotools import read_tmy3
+from pvlib.solarposition import get_solarposition
+from pvlib import irradiance
+import pvlib
+
+# For this example we use the Greensboro, North Carolina, TMY3 file which is
+# in the pvlib data directory. TMY3 are made from the median months from years
+# of data measured from 1990 to 2010. Therefore we change the timestamps to a
+# common year, 1990.
+DATA_DIR = pathlib.Path(pvlib.__file__).parent / 'data'
+greensboro, metadata = read_tmy3(DATA_DIR / '723170TYA.CSV', coerce_year=1990)
+
+# Many of the diffuse fraction estimation methods require the "true" zenith, so
+# we calculate the solar positions for the 1990 at Greensboro, NC.
+solpos = get_solarposition(
+    greensboro.index, latitude=metadata['latitude'],
+    longitude=metadata['longitude'], altitude=metadata['altitude'],
+    pressure=greensboro.Pressure*100, temperature=greensboro.DryBulb)
```

Comment:
Maybe add a comment explaining the `*100` on the pressure (conversion from millibar to Pa, I assume)